benchmarking/simulation/simulation.py

```diff
@@ -33,6 +34,14 @@ def __init__(
         self.team_sets = []
 
     def run(self, num_runs: int) -> SimulationArtifact:
+        if self.settings.cache_key:
```

Comment:
maybe make a ticket for the whole `cache_key="folder/folder/name` where it would make those nested folders automatically